werkit/compute/test_manager.py

```diff
@@ -112,12 +112,41 @@ def test_manager_passes_keyboard_interrupt() -> None:
             raise KeyboardInterrupt()
 
 
-def test_manager_passes_value_error_when_no_message_key_present() -> None:
+def test_manager_passes_value_error_and_skips_body_when_input_message_has_no_message_key_and_fails_validation() -> (
+    None
+):
+    from unittest.mock import Mock
+
+    mock = Mock()
+
+    with pytest.raises(
+        ValueError, match="Input message is missing `message_key` property"
+    ):
+        with create_manager(schema=schema, input_message=math.pi):
+            mock()
+
+    mock.assert_not_called()
+
+
+def test_manager_passes_value_error_and_skips_body_when_input_message_has_no_message_key_and_passes_validation() -> (
```

Comment:
This rather surprising behavior – that the body is run – is the bug being fixed here.